src/server/dfly_main.cc

```diff
@@ -860,9 +870,92 @@ Usage: dragonfly [FLAGS]
     AcceptServer acceptor(pool.get(), &fb2::std_malloc_resource, true);
     acceptor.set_back_log(absl::GetFlag(FLAGS_tcp_backlog));
 
-    dfly::RunEngine(pool.get(), &acceptor);
+#ifdef USE_AFL
+    // AFL++ persistent mode: server in thread, fuzzing in main
+    std::thread server_thread([&]() {
+      dfly::RunEngine(pool.get(), &acceptor);
+      pool->Stop();
+    });
+
+    // Wait for server
+    bool server_ready = false;
```

Comment:
1. can you please factor out into a dedicated function?
2. Also please add comments in that function explaining what it does.

Reply:
Maybe even more than a single dedication function...